### DISCUSSION TITLE
Tradeoffs between Linux and Windows for production installation

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#390] [General]
Which OS and architecture do you recommend for a production installation of Leaf, Linux or Windows?

Leaf will be deployed in a VM on a Linux HPC cluster and will be used to access an OHDSI CDM. A Windows VM that runs SQL Server will store the CDM and the LeafDB. We plan to use SAML 2 authentication.

If all other factors are equal, we'd prefer to run Linux in the Leaf VM, as Windows VMs are harder to configure and maintain on the cluster. 

Your thoughts?

Thanks
Arthur

**Comment by ndobb** [comment]
That makes sense, @artgoldberg . Our team at UW prefers Linux as well for production Leaf deployments, as we have more in-house expertise with Apache+Shibboleth/SAML2 application deployments and also have moved away from Windows VMs for the reasons you mention.

Along those lines, if you pursue a Linux deployment we can more readily advise and share configuration files, etc.

Please let me know if you have other questions.

Best,
-nic

**Reply by artgoldberg** [reply]
Hi @ndobb . I appreciate your quick and helpful reply.
Regards
Arthur
